hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -110,6 +110,161 @@ public static function findById($id) {
+    // ];
+    // If there are two (or more) Contracts starting on the same day
+    // then we pick only the one with latest End date.
+    foreach ($contracts['values'] as $contract) {
+      $details = civicrm_api3('HRJobDetails', 'getsingle', array(
+        'sequential' => 1,
+        'jobcontract_id' => (int)$contract['id'],
+      ));
+      if (empty($details['period_end_date'])) {
+        $dates[$details['period_start_date']] = null;
+        break;
+      }
+      if (!empty($dates[$details['period_start_date']])) {
+        if ($details['period_end_date'] > $dates[$details['period_end_date']]) {
+          $dates[$details['period_start_date']] = $details['period_end_date'];
+        }
+        continue;
+      }
+      $dates[$details['period_start_date']] = $details['period_end_date'];
+    }
+
+    // Sorting $dates array by keys.
+    ksort($dates);
+
+    // Calculate Service Start Date and Service End Date.
+    foreach ($dates as $startDate => $endDate) {
+      if (!$serviceStartDate) {
+        $serviceStartDate = $startDate;
+      }
+      if (!$serviceEndDate) {
+        $serviceEndDate = $endDate;
+      }
+      if ($startDate <= self::sumDateAndBreak($serviceEndDate, $break)) {
+        $serviceEndDate = $endDate;
+      } else {
+        $serviceStartDate = $startDate;
+        $serviceEndDate = $endDate;
+      }
+      if (!$serviceEndDate) {
+        break;
+      }
+    }
+
+    // Restrict $serviceEndDate to date defined at the top,
+    // so we won't get an infinite Service length.
+    if (!$serviceEndDate ||  $serviceEndDate > $date) {
+      $serviceEndDate = $date;
+    }
+
+    // If the latest Contract has ended more than $break days ago, we return 0.
+    if ($date > self::sumDateAndBreak($serviceEndDate, $break)) {
+      return 0;
+    }
+
+    // Final Length of Service calculations by difference of both dates.
+    $dStart = new DateTime($serviceStartDate);
+    $dEnd  = new DateTime($serviceEndDate);
+    $dDiff = $dStart->diff($dEnd);
+    return $dDiff->days;
+  }
+
+  /** Calculate a new Date which is sum of given Date and number of Break days.
+   * Returns null if given date is null.
+   * 
+   * @param string $date
+   * @param int $break
+   * @return string
+   */
+  protected static function sumDateAndBreak($date, $break) {
+    if (!$date) {
+      return null;
+    }
+    $newDate = new DateTime($date);
+    $newDate->add(new DateInterval('P' . $break . 'D'));
+    return $newDate->format('Y-m-d');
+  }
+  
+  /**
+   * Update Length of Service for specific Contact.
+   * 
+   * @return bool
+   */
+  public static function updateLengthOfService($contactId) {
+    // Get Length of Service's Custom Field ID.
+    $cgID = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_CustomGroup', 'Contact_Length_Of_Service', 'id', 'name');
+    $cf = civicrm_api3('CustomField', 'getsingle', array('custom_group_id' => $cgID, 'name' => 'Length_Of_Service'));
+    $cfID = $cf['id'];
```

Comment:
Not sure what's happening here, it looks like you are overriding the `$cfID` that you had just defined to lines above?